core/src/main/java/jenkins/model/ProjectNamingStrategy.java

```diff
@@ -66,11 +66,27 @@ public static DescriptorExtensionList<ProjectNamingStrategy, ProjectNamingStrate
      *            the name given from the UI
      * @throws Failure
      *             if the user has to be informed about an illegal name, forces the user to change the name before submitting. The message of the failure will be presented to the user.
+     * @deprecated Use {@link #checkName(String, String)}
      */
+    @Deprecated  
     public void checkName(String name) throws Failure {
         // no op
     }
 
+    /**
+     * Called when creating a new job.
+     *
+     * @param parentName
+     *            the full name of the parent ItemGroup
+     * @param name
+     *            the name given from the UI
+     * @throws Failure
+     *             if the user has to be informed about an illegal name, forces the user to change the name before submitting. The message of the failure will be presented to the user.
+     */
+    public void checkName(String parentName, String name) throws Failure {
+      checkName(name);
+    }
```

Comment:
To keep the same indentation in the file

```suggestion
    public void checkName(String parentName, String name) throws Failure {
        checkName(name);
    }
```